src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Order;
+import com.internet.shop.storage.Storage;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    @Override
+    public Order create(Order order) {
+        Storage.addOrder(order);
+        return order;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        return Storage.orders.stream().filter(order -> order.getId().equals(id)).findFirst();
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return Storage.orders.stream()
+                .filter(order -> order.getUserId().equals(userId))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Order> getAll() {
+        return Storage.orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        IntStream.range(0, Storage.orders.size())
+                .filter(i -> Storage.orders.get(i).getId().equals(order.getId()))
+                .forEach(item -> Storage.orders.set(item, order));
```

Comment:
```suggestion
                .forEach(i -> Storage.orders.set(i, order));
```
it's index, not item